### ISSUE TITLE
[Bug Report] `loadbalancer` sub-module fails when specifying `public_ip_resource_group_name` for existing Public IP resource for FrontEnd IP configuration

### ISSUE THREAD

**Issue by jinkang23** [PaloAltoNetworks/terraform-azurerm-swfw-modules#51]
### Describe the bug

Load balancer sub-module on line #51 has a typo that is causing an error during `terraform plan` when specifying an existing Public IP resource. 

Output of `terraform plan` describing the error:  
```hcl
╷
│ Error: Unsupported attribute
│
│   on modules/palo-alto-nva-ngfw/loadbalancer/main.tf line 51, in data "azurerm_public_ip" "this":
│   51:   resource_group_name = coalesce(each.value.public_ip_resource_group, var.resource_group_name)
│     ├────────────────
│     │ each.value is object with 9 attributes
│
│ This object does not have an attribute named "public_ip_resource_group".
╵
Operation failed: failed running terraform plan (exit 1)
```   

Template of the error: [terraform-azurerm-swfw-modules](https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/tree/main)/[modules](https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/tree/main/modules)/[loadbalancer](https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/tree/main/modules/loadbalancer)
/main.tf. 

```hcl
# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/public_ip
data "azurerm_public_ip" "this" {
  for_each = { for k, v in var.frontend_ips : k => v if !v.create_public_ip && v.public_ip_name != null }

  name                = each.value.public_ip_name
  resource_group_name = coalesce(each.value.public_ip_resource_group, var.resource_group_name)
}
```  

This `data` source should be updated as follows:  

`coalesce` function's first element should be updated from `each.value.public_ip_resource_group` to `each.value.public_ip_resource_group_name`.  

```hcl
# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/public_ip
data "azurerm_public_ip" "this" {
  for_each = { for k, v in var.frontend_ips : k => v if !v.create_public_ip && v.public_ip_name != null }

  name                = each.value.public_ip_name
  resource_group_name = coalesce(each.value.public_ip_resource_group_name, var.resource_group_name)
}
```   

### Module Version

v3.0.0

### Terraform version

1.8.5

**Comment by acelebanski** [comment]
Hello @jinkang23, good catch! This will be resolved by PR #53. Thank you!

**Comment by jinkang23** [comment]
Great, Thank you @acelebanski !
